frontend/src/components/categoryList/categoryList.js

```diff
@@ -1,24 +1,102 @@
 import categoryService from '../../services/category.service';
 import {useEffect, useState} from "react";
+import {Link} from 'react-router-dom';
+import plus from '../../images/Plus.png'
+import minus from "../../images/Minus.png"
 
-function CategoryList() {
+function CategoryList({filter}) {
     const [categories, setCategories] = useState();
+    let show = useRef([]);
+    let mainContainer = "list"
+    let listContainer = "listItem"
+    let previousCategory = useRef(-1);
+    const [sign, setSign] = useState(-1);
+    let prodCat = useRef("");
+
+    function showSubcategories(event) {
+        let showArray = [...show.current]
+        let index = event.target.id - 1;
+        if (event.target.src === plus) {
+            event.target.src = minus;
+            showArray[index] = true
+        } else {
+            event.target.src = plus;
+            showArray[index] = false;
+        }
+        show.current = showArray;
+    }
+
     useEffect(() => {
-        categoryService.getAll()
+        categoryService.getCategoriesWithSubcategories()
             .then((response) => {
-                setCategories(response.data);
-            });
-    }, []);
+                setCategories(response.data)
+                if (filter) {
+                    prodCat.current = "PRODUCT "
+                    show.current = new Array(response.data.length).fill(false)
+                    let selected = response.data.find(element => element.category.name.toLowerCase() === filter.toLowerCase());
+                    let showArray = [...show.current];
+                    if (selected) {
+                        if (previousCategory.current !== -1 && previousCategory.current !== selected.category.id - 1) {
+                            showArray[previousCategory.current] = false;
+                            document.getElementsByClassName("showMore")[previousCategory.current].src = plus;
+                        }
+                        showArray[selected.category.id - 1] = true;
+                        show.current = showArray;
+                        if (previousCategory.current !== -1) {
+                            document.getElementsByClassName("showMore")[selected.category.id - 1].src = minus;
+                        } else setSign(1)
+                        previousCategory.current = selected.category.id - 1
+
+                    } else {
+                        if (previousCategory.current !== -1) {
+                            showArray[previousCategory.current] = false;
+                            document.getElementsByClassName("showMore")[previousCategory.current].src = plus;
+                        }
+                    }
+                }
+            })
+        if (sign === 1) {
+            document.getElementsByClassName("showMore")[previousCategory.current].src = minus;
+        }
+
+    }, [filter, sign, show]);
+    if (filter) {
+        mainContainer += " filter";
+        listContainer += "filter"
```

Comment:
I am not sure what these variables represent ? Can you explain shortly in a comment below

Reply:
This variables represent class used since catagory List has 2 different styles.